oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -312,11 +316,39 @@ public AccessToken refreshAccessToken() throws IOException {
     HttpRequest request = requestFactory.buildPostRequest(new GenericUrl(tokenServerUri), content);
     request.setParser(new JsonObjectParser(jsonFactory));
 
-    HttpResponse response;
-    try {
-      response = request.execute();
-    } catch (IOException e) {
-      throw new IOException("Error getting access token for service account: ", e);
+    ExponentialRetryAlgorithm retry =
```

Comment:
ExponentialRetryAlgorithm is thread-safe - we should not construct a new one every time.